monkey/infection_monkey/post_breach/add_user.py

```diff
@@ -0,0 +1,49 @@
+import datetime
+import logging
+import subprocess
+import sys
+from infection_monkey.config import WormConfiguration
+
+LOG = logging.getLogger(__name__)
+
+# Linux doesn't have WindowsError
+try:
+    WindowsError
+except NameError:
+    WindowsError = None
+
+__author__ = 'danielg'
+
+
+class BackdoorUser(object):
+    """
+    This module adds a disabled user to the system.
+    This tests part of the ATT&CK matrix
+    """
+
+    def act(self):
+        LOG.info("Adding a user")
+        if sys.platform.startswith("win"):
+            retval = self.add_user_windows()
+        else:
+            retval = self.add_user_linux()
+        if retval != 0:
+            LOG.warn("Failed to add a user")
+        else:
+            LOG.info("Done adding user")
+
+    @staticmethod
+    def add_user_linux():
+        cmd_line = ['useradd', '-M', '--expiredate',
+                    datetime.datetime.today().strftime('%Y-%m-%d'), '--inactive', '0', '-c', 'MONKEY_USER',
+                    WormConfiguration.ms08_067_remote_user_add]
```

Comment:
I don't understand what happens with passwords here. Looks like you don't specify any password. So the password for our user is empty or does it depend on server's config? Why didn't you add password the same way you did for windows?

Reply:
I want to add a disabled account in both cases. In Windows I'm semi required to provide a password, so I do. In Linux, I can add a user without a password so I do so.

Reply:
Currently I also use the same user/password used in the dead MS08-67 exploit. I guess I need to rename that. :)